redbot/core/core_commands.py

```diff
@@ -1940,6 +1941,357 @@ async def reload(self, ctx: commands.Context, *cogs: str):
+                    "commands, exceeding the {cap} command limit for message commands. "
+                    "Disable some commands first."
+                ).format(count=total_message, cap=CONTEXT_CAP)
+            )
+            return
+        if total_user > CONTEXT_CAP:
+            await ctx.send(
+                _(
+                    "Enabling all application commands from that cog would enable a total of {count} "
+                    "commands, exceeding the {cap} command limit for user commands. "
+                    "Disable some commands first."
+                ).format(count=total_user, cap=CONTEXT_CAP)
+            )
+            return
+
+        # Enable the cogs
+        for name in to_add_slash:
+            await self.bot.enable_app_command(name, discord.AppCommandType.chat_input)
+        for name in to_add_message:
+            await self.bot.enable_app_command(name, discord.AppCommandType.message)
+        for name in to_add_user:
+            await self.bot.enable_app_command(name, discord.AppCommandType.user)
+
+        # Update the tree with the new list of enabled cogs
+        await self.bot.tree.red_check_enabled()
+        await ctx.send(_("Enabled all commands from `{cog_name}`.").format(cog_name=cog_name))
+
+    @slash.command(name="disablecog")
+    async def slash_disablecog(self, ctx: commands.Context, cog_name):
+        """Marks all application commands in a cog as being disabled, preventing them from being added to the bot.
+
+        See a list of cogs with application commands with `[p]slash list`.
+
+        This command does NOT sync the enabled commands with Discord, that must be done manually with `[p]slash sync` for commands to appear in users' clients.
+
+        **Arguments:**
+            - `<cog_name>` - The cog to disable commands from. This argument is case sensitive.
+        """
+        count = 0
+        for name, com in self.bot.tree._global_commands.items():
+            if discord.utils._is_submodule(cog_name, com.module):
+                await self.bot.disable_app_command(name, discord.AppCommandType.chat_input)
+                count += 1
+        for key, com in self.bot.tree._context_menus.items():
+            if discord.utils._is_submodule(cog_name, com.module):
+                name, guild_id, com_type = key
+                await self.bot.disable_app_command(name, discord.AppCommandType(com_type))
+                count += 1
+        if not count:
+            await ctx.send(
+                _(
+                    "Couldn't find any enabled commands from the `{cog_name}` cog. Use `{prefix}slash list` to see all cogs with slash commands."
+                ).format(cog_name=cog_name, prefix=ctx.prefix)
+            )
+            return
+        await self.bot.tree.red_check_enabled()
+        await ctx.send(
+            _("Disabled {count} commands from `{cog_name}`.").format(
```

Comment:
I would like to see this list which commands were disabled to make it more clear which commands were part of the cog.